src/openrct2-ui/windows/Finances.cpp

```diff
@@ -379,14 +379,12 @@ class FinancesWindow final : public Window
         }
 
         // Expenditure / Income values for each month
-        int16_t currentMonthYear = static_cast<int16_t>(gDateMonthsElapsed);
+        uint16_t currentMonthYear = static_cast<uint16_t>(gDateMonthsElapsed);
         for (int32_t i = SummaryMaxAvailableMonth(); i >= 0; i--)
         {
             screenCoords.y = 0;
 
-            int16_t monthyear = currentMonthYear - i;
-            if (monthyear < 0)
-                continue;
```

Comment:
You cannot just remove this check, it is there for a reason.

```
if (i > currentMonthYear)
    continue;
```
should do the trick.

Reply:
Is it even possible to satisfy that condition?
I thought `i` cannot exceed `gDateMonthsElapsed`.

Reply:
I took a look, and you’re right. The if was put in in 2014 to guard against `i > gDateMonthsElapsed`. At the time, `i` was always initialised to 4, rather than `SummaryMaxAvailableMonth()`.